src/code_8048480.c

```diff
@@ -1,19 +1,30 @@
 #include "global.h"
+#include "constants/dungeon_action.h"
 #include "constants/iq_skill.h"
 #include "constants/status.h"
+#include "dungeon_action.h"
 #include "dungeon_entity.h"
+#include "dungeon_util.h"
 #include "dungeon_global_data.h"
 #include "move_effects_target.h"
+#include "dungeon_capabilities.h"
 #include "dungeon_pokemon_attributes.h"
 #include "dungeon_random.h"
 #include "number_util.h"
 #include "moves.h"
 #include "code_8077274_1.h"
+#include "code_80521D0.h"
+#include "pokemon.h"
 #include "position.h"
 #include "status.h"
 
+extern u8 gAvailablePokemonNames[];
+extern u8 gUnknown_202DE58[];
+
+extern s16 gTypeGummiIQBoost[0x12][NUMBER_OF_GUMMIS];
```

Comment:
0x12 is the number of elemental types, which can use `NUM_TYPES` from `constants/type.h`.